code/modules/mob/living/carbon/give.dm

```diff
@@ -1,52 +1,186 @@
+	to_chat(user, "<span class='info'>You offer [I] to [receiver].</span>")
+	qdel(src)
+
+
+/**
+ * # Take Item alert
+ *
+ * Alert which appears for a user when another player is attempting to offer them an item.
+ * The user can click the alert to accept, or simply do nothing to not take the item.
+ */
+/obj/screen/alert/take_item
+	name = "Take Item"
+	desc = "someone wants to hand you an item!"
+	icon_state = "template"
+	timeout = 10 SECONDS
+	/// UID of the mob offering the receiver an item.
+	var/giver_UID
+	/// UID of the mob who has this alert.
+	var/receiver_UID
+	/// UID of the item being given.
+	var/item_UID
+	/// If the receiver succesfully took the item or not.
+	var/item_taken = FALSE
+
+
+/obj/screen/alert/take_item/Initialize(mapload, mob/living/giver, mob/living/receiver, obj/item/I)
+	. = ..()
+	desc = "[giver] wants to hand you \a [I]. Click here to accept it!"
+	giver_UID = giver.UID()
+	receiver_UID = receiver.UID()
+	item_UID = I.UID()
+	giver.apply_status_effect(STATUS_EFFECT_OFFERING_ITEM)
+	add_overlay(icon(I.icon, I.icon_state, SOUTH))
+	add_overlay("alert_flash")
+	RegisterSignal(I, list(COMSIG_ITEM_EQUIPPED, COMSIG_ITEM_DROPPED), .proc/cancel_give)
+	RegisterSignal(giver, list(SIGNAL_ADDTRAIT(TRAIT_HANDS_BLOCKED), COMSIG_CARBON_SWAP_HANDS), .proc/cancel_give)
+	// If any one of these atoms should be deleted, we need to cancel everything. Also saves having to do null checks before interacting with these atoms.
+	RegisterSignal(I, COMSIG_PARENT_QDELETING, /datum/proc/signal_qdel)
+	RegisterSignal(giver, COMSIG_PARENT_QDELETING, /datum/proc/signal_qdel)
+	RegisterSignal(receiver, COMSIG_PARENT_QDELETING, /datum/proc/signal_qdel)
+
+
+/obj/screen/alert/take_item/Destroy()
+	var/mob/living/giver = locateUID(giver_UID)
+	giver.remove_status_effect(STATUS_EFFECT_OFFERING_ITEM)
+	if(!item_taken)
+		unregister_signals()
+	return ..()
+
+
+/obj/screen/alert/take_item/proc/unregister_signals()
+	var/mob/living/giver = locateUID(giver_UID)
+	var/obj/item/I = locateUID(item_UID)
+	UnregisterSignal(I, list(COMSIG_PARENT_QDELETING, COMSIG_ITEM_EQUIPPED, COMSIG_ITEM_DROPPED))
+	UnregisterSignal(giver, list(COMSIG_PARENT_QDELETING, SIGNAL_ADDTRAIT(TRAIT_HANDS_BLOCKED), COMSIG_CARBON_SWAP_HANDS))
+	UnregisterSignal(locateUID(receiver_UID), COMSIG_PARENT_QDELETING)
 
-	if(target.incapacitated() || usr.incapacitated() || target.client == null)
+
+/obj/screen/alert/take_item/proc/cancel_give()
+	SIGNAL_HANDLER
+	var/mob/living/giver = locateUID(giver_UID)
+	to_chat(giver, "<span class='warning'>You need to keep the item in your active hand if you want to hand it to someone!</span>")
+	to_chat(locateUID(receiver_UID), "<span class='warning'>[giver] seems to have given up on giving you [locateUID(item_UID)].</span>")
+	qdel(src)
+
+
+/obj/screen/alert/take_item/Click(location, control, params)
+	var/mob/living/receiver = locateUID(receiver_UID)
+	if(receiver.stat != CONSCIOUS)
+		return
+	var/obj/item/I = locateUID(item_UID)
```

Comment:
Again, will this still work sanely if the item has been deleted?

Reply:
This one I did have monitoring the item for `COMSIG_PARENT_QDELETING`. So as soon as the item is qdel'd this alert is qdel'd as well.